src/details/ArborX_DetailsDistributedTreeNearestHelpers.hpp

```diff
@@ -0,0 +1,271 @@
+struct AccessTraits<
+    Details::WithinDistanceFromPredicates<Predicates, Distances>, PredicatesTag>
+{
+  using Predicate = typename Predicates::value_type;
+  using Geometry =
+      std::decay_t<decltype(getGeometry(std::declval<Predicate const &>()))>;
+  using Self = Details::WithinDistanceFromPredicates<Predicates, Distances>;
+
+  using memory_space = typename Predicates::memory_space;
+  using size_type = decltype(std::declval<Predicates const &>().size());
+
+  static KOKKOS_FUNCTION size_type size(Self const &x)
+  {
+    return x.predicates.size();
+  }
+  template <class Dummy = Geometry,
+            std::enable_if_t<std::is_same_v<Dummy, Geometry> &&
+                             std::is_same_v<Dummy, Point>> * = nullptr>
+  static KOKKOS_FUNCTION auto get(Self const &x, size_type i)
+  {
+    auto const point = getGeometry(x.predicates(i));
+    auto const distance = x.distances(i);
+    return intersects(Sphere{point, distance});
+  }
+  template <class Dummy = Geometry,
+            std::enable_if_t<std::is_same_v<Dummy, Geometry> &&
+                             std::is_same_v<Dummy, Box>> * = nullptr>
+  static KOKKOS_FUNCTION auto get(Self const &x, size_type i)
+  {
+    auto box = getGeometry(x.predicates(i));
+    auto &min_corner = box.minCorner();
+    auto &max_corner = box.maxCorner();
+    auto const distance = x.distances(i);
+    for (int d = 0; d < 3; ++d)
+    {
+      min_corner[d] -= distance;
+      max_corner[d] += distance;
+    }
+    return intersects(box);
+  }
+  template <class Dummy = Geometry,
+            std::enable_if_t<std::is_same_v<Dummy, Geometry> &&
+                             std::is_same_v<Dummy, Sphere>> * = nullptr>
+  static KOKKOS_FUNCTION auto get(Self const &x, size_type i)
+  {
+    auto const sphere = getGeometry(x.predicates(i));
+    auto const distance = x.distances(i);
+    return intersects(Sphere{sphere.centroid(), distance + sphere.radius()});
+  }
+  template <
+      class Dummy = Geometry,
+      std::enable_if_t<std::is_same_v<Dummy, Geometry> &&
+                       std::is_same_v<Dummy, Experimental::Ray>> * = nullptr>
+  static KOKKOS_FUNCTION auto get(Self const &x, size_type i)
+  {
+    auto const ray = getGeometry(x.predicates(i));
+    return intersects(ray);
+  }
+};
+
+namespace Details
+{
+
+template <typename Tree, typename Callback, typename OutValue, bool UseValues>
+struct CallbackWithDistance
+{
+  Tree _tree;
+  Callback _callback;
+
+  template <typename ExecutionSpace>
+  CallbackWithDistance(ExecutionSpace const &, Tree const &tree,
+                       Callback const &callback)
+      : _tree(tree)
+      , _callback(callback)
+  {}
+
+  template <typename Query, typename Value, typename Output>
+  KOKKOS_FUNCTION void operator()(Query const &query, Value const &value,
+                                  Output const &out) const
+  {
+    if constexpr (UseValues)
+    {
+      OutValue out_value;
+      [[maybe_unused]] int count = 0;
```

Comment:
Why did you add the `[[maybe_unused]]` attribute here?
It is being incremented below.  Were you actually seeing warning with `NDEBUG` defined?

Reply:
The compiler [complained](https://cloud.cees.ornl.gov/jenkins-ci/job/ArborX/job/PR-1075/2/gcc/folder.1897384215/) about it.